src/coreWebVitals.test.ts

```diff
@@ -0,0 +1,219 @@
+
+		expect(mockCallback).toBeCalledTimes(1);
+		expect(mockCallback).toBeCalledWith(false);
+	});
+
+	it('triggers the callback with `true` if metrics will be sent', () => {
+		const mockCallback = jest.fn();
+		initCoreWebVitals(
+			{ browserId: 'abc', pageViewId: '123' },
+			mockCallback,
+		);
+
+		forceSendMetrics();
+		global.dispatchEvent(new Event('pagehide'));
+
+		expect(mockCallback).toBeCalledTimes(1);
+		expect(mockCallback).toHaveBeenLastCalledWith(true);
+	});
+
+	it('does not trigger a callback if none is passed', () => {
+		const mockCallback = jest.fn(); // won’t be used
+		const mockAddEventListener = jest.spyOn(global, 'addEventListener');
+		initCoreWebVitals({ browserId: 'abc', pageViewId: '123' });
+
+		setVisibilityState('hidden');
+		global.dispatchEvent(new Event('visibilitychange'));
+		global.dispatchEvent(new Event('pagehide'));
+
+		expect(mockCallback).not.toHaveBeenCalled();
+		expect(mockAddEventListener).toHaveBeenCalledTimes(2);
+	});
+});
+
+describe('roundWithDecimals', () => {
+	it.each([
+		[1, 3, 3],
+		[1, 10.0, 10],
+		[1, 10.3, 10.3],
+		[1, 2.5, 2.5],
+		[3, 0.001_234, 0.001],
+		[4, 0.001_234, 0.001_2],
+		[5, 100.102_030_405_060_708_090, 100.102_03],
+		[6, 12345.000_001_2, 12345.000_001],
+		[9, 199.001_002_003_456, 199.001_002_003],
+	])('With precision %s, %f becomes %f', (precision, before, after) => {
+		expect(roundWithDecimals(before, precision)).toBe(after);
+	});
+
+	it('should handle default precision = 6', () => {
+		const [before, after] = [12345.000_001_2, 12345.000_001];
+		expect(roundWithDecimals(before)).toBe(after);
+		expect(roundWithDecimals(before, 6)).toBe(after);
+	});
+});
+
+describe('sendData', () => {
+	beforeEach(() => {
+		jest.resetModules();
+		window.location.hash = '';
+		resetShouldForceMetrics();
+	});
+
+	const browserId = String(defaultCoreWebVitalsPayload.browser_id);
+	const pageViewId = String(defaultCoreWebVitalsPayload.page_view_id);
+
+	it('should send data if in sample', () => {
+		mockMath.mockReturnValueOnce(0.1 / 100);
+		initCoreWebVitals({ browserId, pageViewId });
+
+		expect(sendData()).toBe(true);
+	});
+
+	it('should not send data if not in sample', () => {
+		initCoreWebVitals({ browserId, pageViewId });
+
+		expect(sendData()).toBe(false);
+	});
+
+	it('should send data if not in sample but forced via init', () => {
+		initCoreWebVitals({
+			browserId,
+			pageViewId,
+			forceSendMetrics: true,
+		});
+
+		expect(sendData()).toBe(true);
+	});
+
+	it('should send data if not in sample but forced via hash', () => {
+		window.location.hash = '#forceSendMetrics';
+		initCoreWebVitals({ browserId, pageViewId });
+
+		expect(sendData()).toBe(true);
+	});
+
+	it('should send data if forced asynchronously', () => {
+		initCoreWebVitals({ browserId, pageViewId });
+
+		expect(sendData()).toBe(false);
+
+		forceSendMetrics();
+		expect(sendData()).toBe(true);
+	});
+
+	it('should use CODE URL by default', () => {
```

Comment:
Should it? Is there a way to make specifying the endpoint required so there's no default? I feel like if you are falling back on a default something has gone wrong